### ISSUE TITLE
KeyError: 'snippets/facet_list.html'

### ISSUE THREAD

**Issue by pwalsh** [ckan/ckan#2808]
# Background

I'm seeing a range of simple exceptions coming out of datahub.io (running CKAN 2.4). These errors are not critical, but should be addressed and are probably easy "good for contribution" tasks.
# Issue

```
KeyError: 'snippets/facet_list.html'
  File "raven/middleware.py", line 35, in __call__
    iterable = self.application(environ, start_response)
  File "webob/dec.py", line 147, in __call__
    resp = self.call_func(req, *args, **self.kwargs)
  File "webob/dec.py", line 208, in call_func
    return self.func(req, *args, **kwargs)
  File "fanstatic/publisher.py", line 234, in __call__
    return request.get_response(self.app)
  File "webob/request.py", line 1053, in get_response
    application, catch_exc_info=False)
  File "webob/request.py", line 1022, in call_application
    app_iter = application(self.environ, start_response)
  File "webob/dec.py", line 147, in __call__
    resp = self.call_func(req, *args, **self.kwargs)
  File "webob/dec.py", line 208, in call_func
    return self.func(req, *args, **kwargs)
  File "fanstatic/injector.py", line 54, in __call__
    response = request.get_response(self.app)
  File "webob/request.py", line 1053, in get_response
    application, catch_exc_info=False)
  File "webob/request.py", line 1022, in call_application
    app_iter = application(self.environ, start_response)
  File "ckan/config/middleware.py", line 389, in inner
    result = application(environ, start_response)
  File "beaker/middleware.py", line 73, in __call__
    return self.app(environ, start_response)
  File "beaker/middleware.py", line 155, in __call__
    return self.wrap_app(environ, session_start_response)
  File "routes/middleware.py", line 131, in __call__
    response = self.app(environ, start_response)
  File "pylons/wsgiapp.py", line 125, in __call__
    response = self.dispatch(controller, environ, start_response)
  File "pylons/wsgiapp.py", line 324, in dispatch
    return controller(environ, start_response)
  File "ckan/lib/base.py", line 338, in __call__
    res = WSGIController.__call__(self, environ, start_response)
  File "pylons/controllers/core.py", line 221, in __call__
    response = self._dispatch_call()
  File "pylons/controllers/core.py", line 172, in _dispatch_call
    response = self._inspect_call(func)
  File "pylons/controllers/core.py", line 107, in _inspect_call
    result = self._perform_call(func, args)
  File "ckan/controllers/group.py", line 230, in read
    extra_vars={'group_type': group_type})
  File "ckan/lib/base.py", line 217, in render
    loader_class=loader_class)
  File "pylons/templating.py", line 249, in cached_template
    return render_func()
  File "ckan/lib/base.py", line 155, in render_template
    return render_jinja2(template_name, globs)
  File "ckan/lib/base.py", line 103, in render_jinja2
    template = env.get_template(template_name)
  File "jinja2/environment.py", line 719, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "jinja2/environment.py", line 695, in _load_template
    self.cache[name] = template
  File "jinja2/utils.py", line 473, in __setitem__
    del self._mapping[self._popleft()]
```

**Comment by rossjones** [comment]
This looks like it is a problem with the sentry-middleware rather than a CKAN specific problem.  And presumably a problem when it can't find that specific snippet.

Is the sentry integration in a different repo somewhere?

**Comment by pwalsh** [comment]
Hey @rossjones 

Yes, it is here: https://github.com/okfn/ckanext-sentry

I have more errors to report, and I was trying to ensure I only put errors here that looked to me came from ckan core. Perhaps I should open all issues on the datapackage extension repo first and then we can migrate over the relevant errors to `ckan/ckan`

**Comment by rossjones** [comment]
Actually I think the raven middleware might be hiding another problem, so a bit of a red-herring.  Looks like it can't find the snippets/facet_list.html file, but it should be there from core.  I'm trying to replicate this on ckanext-datahub, do you have the URL that was requested that triggered it?

**Comment by pwalsh** [comment]
Yes, sure:

http://datahub.io/organization/196c4192-39bb-4644-9992-819b8411804c

**Comment by rossjones** [comment]
I can't replicate this at all, and so I suspect it is a datahub specific error - but I can't even replicate it with ckanext-datahub, where is always successfully finds and loads that snippet.

**Comment by pwalsh** [comment]
@rossjones actually looking closer, this, like many other errors I'm getting on datahub.io, is triggered by search engine crawlers. 

Without diving into the CKAN code base at all, but based on similar experiences in ~large codebases I've deployed in the past, I'd wager that these "logic in templates" derived errors are actually cases where templates are written to expect a certain user state (e.g.: authenticated, or, certain session keys) and there are no defensive checks for users that don't have such state (like crawlers, and anon users in certain conditions).

**Comment by MrkGrgsn** [comment]
I've just seen the same error with CKAN 2.5.4 (also reported by Sentry). I suspect it's caused by https://github.com/pallets/jinja/issues/8 which was resolved in jinja2 2.7 released in May 2013.  Perhaps a requirements upgrade is due?

**Comment by kmbn** [comment]
We decided to close old issues that are not actively worked on so that we can focus our effort and attention on issues affecting the current versions of CKAN.

If this issue is still affecting the version of CKAN you're working with now, please feel free to comment or reopen the issue.

If you do reopen this issue, please update it with new details. One reason it might not have been resolved in the past is that it wasn't clear how a contributor could address the issue.
